src/commands/api/request/rest.ts

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { EOL } from 'node:os';
+import { createWriteStream } from 'node:fs';
+import got, { Headers } from 'got';
+import type { AnyJson } from '@salesforce/ts-types';
+import { ProxyAgent } from 'proxy-agent';
+import { Flags, SfCommand } from '@salesforce/sf-plugins-core';
+import { Messages, Org, SfError } from '@salesforce/core';
+import { Args } from '@oclif/core';
+import ansis from 'ansis';
+
+Messages.importMessagesDirectoryFromMetaUrl(import.meta.url);
+const messages = Messages.loadMessages('@salesforce/plugin-api', 'rest');
+
+export class Rest extends SfCommand<void> {
+  public static readonly summary = messages.getMessage('summary');
+  public static readonly examples = messages.getMessages('examples');
+  public static state = 'beta';
+  public static enableJsonFlag = false;
+  public static readonly flags = {
+    // TODO: getting a false positive from this eslint rule.
+    // summary is already set in the org flag.
+    // eslint-disable-next-line sf-plugin/flag-summary
+    'target-org': Flags.requiredOrg({
+      helpValue: 'username',
+    }),
+    include: Flags.boolean({
+      char: 'i',
+      summary: messages.getMessage('flags.include.summary'),
+      default: false,
+      exclusive: ['stream-to-file'],
+    }),
+    method: Flags.option({
+      options: ['GET', 'POST', 'PUT', 'PATCH', 'HEAD', 'DELETE', 'OPTIONS', 'TRACE'] as const,
+      summary: messages.getMessage('flags.method.summary'),
+      char: 'X',
+      default: 'GET',
+    })(),
+    header: Flags.string({
+      summary: messages.getMessage('flags.header.summary'),
+      helpValue: 'key:value',
+      char: 'H',
+      multiple: true,
+    }),
+    'stream-to-file': Flags.string({
+      summary: messages.getMessage('flags.stream-to-file.summary'),
+      helpValue: 'Example: report.xlsx',
+      char: 'S',
+      exclusive: ['include'],
+    }),
+    body: Flags.string({
+      summary: messages.getMessage('flags.body.summary'),
+      allowStdin: true,
+      helpValue: 'file',
+    }),
+  };
+
+  public static args = {
+    endpoint: Args.string({
+      description: 'Salesforce API endpoint',
+      required: true,
+    }),
+  };
+
+  private static getHeaders(keyValPair: string[]): Headers {
```

Comment:
another benefit of having a fn is that you could put it as the `parse` prop on the `--header` flag so that if people have bad headers it fails sooner (avoiding org re-auth, etc)